src/projections/isea.cpp

```diff
@@ -1105,5 +1108,456 @@ PJ *PJ_PROJECTION(isea) {
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE.
+*/
+
+/* The ISEA projection a projects a sphere on the icosahedron. Thereby the size
+ * of areas mapped to the icosahedron are preserved. Angles and distances are
+ * however slightly distorted. The angular distortion is below 17.27 degree, and
+ * the scale variation is less than 16.3 per cent.
+ *
+ * The projection has been proposed and has been described in detail by:
+ *
+ * John P. Snyder: An equal-area map projection for polyhedral globes.
+ * Cartographica, 29(1), 10–21, 1992. doi:10.3138/27H7-8K88-4882-1752
+ *
+ * Another description and improvements can be found in:
+ *
+ * Erika Harrison, Ali Mahdavi-Amiri, and Faramarz Samavati: Optimization of
+ * inverse Snyder polyhedral projection. International Conference on Cyberworlds
+ * 2011. doi:10.1109/CW.2011.36
+ *
+ * Erika Harrison, Ali Mahdavi-Amiri, and Faramarz Samavati: Analysis of inverse
+ * Snyder optimizations. In: Marina L. Gavrilova, and C. J. Kenneth Tan (Eds):
+ * Transactions on Computational Science XVI. Heidelberg, Springer, 2012. pp.
+ * 134–148. doi:10.1007/978-3-642-32663-9_8
+ */
+
+struct GeoPoint {
+    double lat, lon;
+}; // In radians
+
+#define Pi 3.1415926535897932384626433832795028841971
+
+#define Degrees(x) ((x)*Pi / 180)
+
+#define wgs84InvFlattening 298.257223563
+#define wgs84Major 6378137.0 // Meters
+#define wgs84Minor                                                             \
+    (wgs84Major - (wgs84Major / wgs84InvFlattening)) // 6356752.3142451792955399
+
+// #define phi                ((1 + sqrt(5)) / 2)
+// radius
+// #define a2                 (wgs84Major * wgs84Major)
+// #define c2                 (wgs84Minor * wgs84Minor)
+
+// #define eccentricity       0.0818191908426 // sqrt(1.0 - (c2/a2));
+// #define log1pe_1me         0.1640050079196   // log((1+e)/(1-e)))
+// #define S                  (Pi * (2 * a2 + c2/eccentricity * log1pe_1me))
+#define earthAuthalicRadius                                                    \
+    6371007.18091875 // sqrt(S / (4 * Pi)); // R -- authalic sphere radius for
+                     // WGS84 [6371007.1809184747 m]
+#define R2 (earthAuthalicRadius * earthAuthalicRadius) // R^2
+#define RprimeOverR                                                            \
+    0.9103832815095032 // (1 / (2 * sqrt(5)) + 1 / 6.0) * sqrt(Pi * sqrt(3)); //
+                       // R' / R
+#define Rprime (RprimeOverR * earthAuthalicRadius) // R'
+
+#include <algorithm>
+
+#define Min std::min
+#define Max std::max
+
+#define inf std::numeric_limits<double>::infinity()
+
+// distortion
+// static double maximumAngularDistortion = 17.27;
+// static double maximumScaleVariation = 1.163;
+// static double minimumScaleVariation = .860;
+
+// Vertices of dodecahedron centered in icosahedron faces
+#define E                                                                      \
```

Comment:
the E_RAD constant could be reused

Reply:
The original `E_RAD` constant is not precise enough. Can I fix it and the original tests?

```diff
-/* 52.62263186 */
-#define E_RAD 0.91843818702186776133
+// Latitude of center of top icosahedron faces
+// atan((3 + sqrt(5)) / 4) = 52.6226318593487 degrees
+#define E_RAD 0.91843818701052843323

 accept  2 -1
-expect  -1097074.948264795 3233611.728585708
+expect  -1097074.948260187404 3233611.728216679767
 accept  -2 1
 expect  -1575486.353641554 3442168.342028188
 accept  -2 -1
-expect  -1575486.353880283 3234352.695594706
+expect  -1575486.353875697823 3234352.695233523380
```

Reply:
Changes applied.